core/src/subgraph/runner.rs

```diff
@@ -475,13 +477,26 @@ where
 
         for info in created_data_sources {
             // Try to instantiate a data source from the template
-            let data_source = match DataSource::from_template_info(info) {
-                Ok(ds) => ds,
-                Err(e @ DataSourceCreationError::Ignore(..)) => {
-                    warn!(self.logger, "{}", e.to_string());
-                    continue;
+
+            let data_source = {
+                let res = match info.template {
+                    DataSourceTemplate::Onchain(_) => C::DataSource::from_template_info(info)
+                        .map(DataSource::Onchain)
+                        .map_err(DataSourceCreationError::from),
+                    DataSourceTemplate::Offchain(_) => offchain::DataSource::from_template_info(
+                        info,
+                        self.ctx.causality_region_next_value(),
+                    )
+                    .map(DataSource::Offchain),
+                };
+                match res {
+                    Ok(ds) => ds,
+                    Err(e @ DataSourceCreationError::Ignore(..)) => {
+                        warn!(self.logger, "{}", e.to_string());
```

Comment:
you are logging this and returning the error with a message. Usually this message would be printed by the handler of the error. As we discussed, I think the Ignore error semantic leads to it being mishandled very easily if someone doesn't know the error is here. I'd rather keep the log message here and return None. If you think you absolutely need the Ignore enum with a message then I'd suggest dropping the warning the error will contain the same message.

Reply:
> you are logging this and returning the error with a message. 

Afaict this is logging and dropping the error after the `continue`.

Reply:
Ah yeah don't mind me, you are correct :)